### PR TITLE
AiSD-14 add Jasmine and add tests for algorithms

### PR DESCRIPTION
**Pull request by mztezi**

### CODE REVIEW

File: src/algorithms/quickSort.spec.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { quickSort } from "./quickSort.js";
+
+describe('mergeSort', () => {
```
Comment (kdryzek) [inline src/algorithms/quickSort.spec.js]:
```suggestion
describe('quickSort', () => {
```

Comment (mztezi) [reply src/algorithms/quickSort.spec.js]: fixed

File: src/algorithms/binarySearch.spec.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import { binarySearch, binarySearchRec } from "./binarySearch.js";
+
+describe('binarySearch', () => {
+    it('should return 1', () => {
+        //assign
+        const arr = [-7, 0, 1, 3, 5];
+        const target1 = 0;
```
Comment (kdryzek) [inline src/algorithms/binarySearch.spec.js]: why targetX here and below?

File: src/algorithms/binarySearch.spec.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        const target2 = 2;
+
+        //act
+        const indexOfTarget2 = binarySearchRec(arr, target2);
+
+        //assert
+        expect(indexOfTarget2).toEqual(-1);
+    })
+
+    it('should return -1', () => {
```
Comment (kdryzek) [inline src/algorithms/binarySearch.spec.js]: duplicate testcase description here and below

File: src/algorithms/binarySearch.spec.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        const target2 = 2;
+
+        //act
+        const indexOfTarget2 = binarySearch(arr, target2);
+
+        //assert
+        expect(indexOfTarget2).toEqual(-1);
+    })
+
+    it('should return -1', () => {
```
Comment (kdryzek) [inline src/algorithms/binarySearch.spec.js]: duplicate testcase

File: src/algorithms/bubbleSort.spec.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import { bubbleSortAlt, bubbleSort } from './bubbleSort.js';
+
+describe('bubbleSort', () => {
+    it('should sort array and return [-6, 0, 1, 4, 7]', () => {
```
Comment (kdryzek) [inline src/algorithms/bubbleSort.spec.js]: bubbleSort doesn't return array, here and below

File: src/algorithms/factorial.spec.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        const testNumber = 0;
+
+        //act
+        const factorialOfTestNumber = factorial(testNumber);
+
+        //assert
+        expect(factorialOfTestNumber).toEqual(1);
+    })
+
+    it('should return 1', () => {
```
Comment (kdryzek) [inline src/algorithms/factorial.spec.js]: duplicate description

File: src/algorithms/fibonacci.spec.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        const nNumber = 2;
+
+        //act
+        const nFibo = fibonacciI(nNumber);
+
+        //assert
+        expect(nFibo).toEqual(1);
+    })
+
+    it('should return 1', () => {
```
Comment (kdryzek) [inline src/algorithms/fibonacci.spec.js]: duplicate description

File: src/algorithms/insertionSort.spec.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import { insertionSort } from "./insertionSort.js";
+
+describe('insertionSort', () => {
+    it('should return empty array', () => {
```
Comment (kdryzek) [inline src/algorithms/insertionSort.spec.js]: insertionSort doesn't return array, here and below

File: src/algorithms/isPowerOfTwo.spec.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        const testNumb = 1;
+
+        //act
+        const answer = isPowerOfTwo(testNumb);
+
+        //assert
+        expect(answer).toBe(true);
+    })
+
+    it('should return true', () => {
```
Comment (kdryzek) [inline src/algorithms/isPowerOfTwo.spec.js]: duplicated description here and below

File: src/algorithms/binarySearch.spec.js (hunk 2 above)
Comment (mztezi) [reply src/algorithms/binarySearch.spec.js]: fixed

File: src/algorithms/binarySearch.spec.js (hunk 3 above)
Comment (mztezi) [reply src/algorithms/binarySearch.spec.js]: discussed and fixed

File: src/algorithms/binarySearch.spec.js (hunk 4 above)
Comment (mztezi) [reply src/algorithms/binarySearch.spec.js]: discussed and fixed

File: src/algorithms/bubbleSort.spec.js (hunk 5 above)
Comment (mztezi) [reply src/algorithms/bubbleSort.spec.js]: discussed, fixed

File: src/algorithms/factorial.spec.js (hunk 6 above)
Comment (mztezi) [reply src/algorithms/factorial.spec.js]: discussed, fixed

File: src/algorithms/fibonacci.spec.js (hunk 7 above)
Comment (mztezi) [reply src/algorithms/fibonacci.spec.js]: discussed, fixed

File: src/algorithms/insertionSort.spec.js (hunk 8 above)
Comment (mztezi) [reply src/algorithms/insertionSort.spec.js]: discussed, fixed

File: src/algorithms/isPowerOfTwo.spec.js (hunk 9 above)
Comment (mztezi) [reply src/algorithms/isPowerOfTwo.spec.js]: discussed, fixed

**Review by kdryzek — APPROVED** [review]

**Pull request merged**